app/uk/gov/hmrc/channelpreferences/controllers/PreferenceController.scala

```diff
@@ -76,4 +83,72 @@ class PreferenceController @Inject()(
+
+  private def doConfirmItsa(passedBackEntityId: String, passedBackItsaId: String, authTokenSaUtr: Option[String])(
+    implicit hc: HeaderCarrier): Future[Result] =
+    entityResolverConnector
+      .resolveBy(passedBackEntityId)
+      .flatMap { entity =>
+        (entity.itsa, entity.saUtr, authTokenSaUtr) match {
+
+          case (Some(entityItsaId), _, _) if entityItsaId == passedBackItsaId =>
+            reply(OK, "itsaId successfully linked to entityId")
+
+          case ( /* entity.itsa */ Some(_), _, _) =>
+            reply(UNAUTHORIZED, s"entityId already has a different itsaId linked to it in entity resolver")
+          // TODO Isn't the following acceptance criteria already included in this case ???
+          /*
+             | Case 1.6 - itsaId is already linked to a different entityId in entity resolver.
+             |
+             |   Given I am a customer who has successfully enrolled in ITSA
+             |   When my itsaId is already linked to a different entityId in entity resolver
+             |   Then my itsaId will not be added (i.e linked to) any entityId
+             |   And an error will be generated
+           */
+
+          case ( /* entity.itsa */ None, _, /* authTokenSaUtr */ None) =>
+            entityResolverConnector
+              .update(entity.copy(itsa = Some(passedBackItsaId)))
+              .flatMap { _ =>
+                reply(OK, "itsaId successfully linked to entityId")
+              }
+
+          case (None, Some(entitySaUtr), Some(authSaUtr)) if entitySaUtr == authSaUtr =>
+            entityResolverConnector
+              .update(entity.copy(itsa = Some(passedBackItsaId)))
+              .flatMap { _ =>
+                reply(OK, "itsaId successfully linked to entityId")
+              }
+
+          case ( /* entity.itsa */ None, /* entity.saUtr */ Some(_), /* authTokenSaUtr */ Some(_)) =>
+            reply(UNAUTHORIZED, "SAUTR in Auth token is different from SAUTR in entity resolver")
+          // TODO Isn't the following acceptance criteria already included in this case ???
+          // Case 1.3 - SAUTR in Auth token is linked to a different entityId in entity resolver
+          //
+          //   Given I am a customer who has successfully enrolled in ITSA
+          //   When SAUTR in Auth token is linked to a different entityId  in entity resolver
+          //   Then my itsaId will not be added (i.e linked to) any entityId
+          //   And an error will be generated
+        }
+      }
+      .recoverWith {
+        case ex: UpstreamErrorResponse if (ex.statusCode == NOT_FOUND) =>
+          /*
+           | Case 3.1 - entityId passed back by ITSA does not exist in entity resolver
+           |
+           |   Given I am a customer who has successfully enrolled in ITSA
+           |   When the entityId passed back does not exist in entity resolver
+           |   Then  my itsaId will not be added to any entityid
+           |   And an error will be generated
```

Comment:
still not sure the comments are useful, seems like more noise that distracts from reading the business logic

Reply:
Ok, fine.
I just removed the comments from within the `recoverWith` block.
The are still 2 comments left. They are marked as:

```
// TODO Need to clarify with Chucks Adichie:
```

as there's still something unclear in the acceptance criteria definitions.